src/main/java/javaslang/collection/Vector.java

```diff
@@ -765,12 +773,20 @@ public int length() {
 
     @Override
     public Vector<T> prepend(T element) {
-        return insert(0, element);
+        final int newIndexShift = indexShift - 1;
+        return new Vector<>(newIndexShift, trie.put(newIndexShift, element));
     }
 
     @Override
     public Vector<T> prependAll(java.lang.Iterable<? extends T> elements) {
-        return insertAll(0, elements);
+        List<T> list = List.ofAll(elements);
```

Comment:
use elements.iterator() (of type java.util.Iterator) instead of List.ofAll(). this will save cycles and memory